docs/user-guide/kubectl-overview.md

```diff
@@ -102,6 +102,7 @@ Resource type    | Abbreviated alias
 `events` |`ev`
 `horizontalpodautoscalers` |`hpa`
 `ingresses` |`ing`
+`initializerconfiguration` |`ic`
```

Comment:
We should stop adding hard coded resources like this and work off of discovery

Reply:
@liggitt Does `apiserver` support resources discovery now.

Reply:
Yes. That is what kubectl actually uses now for shortname expansion

Reply:
So we could remove such hard-coded [`validResources`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/cmd.go#L202) by replacing with resource discovery.

> // TODO: This should be populated using the discovery information from apiserver.

But I found `validResources` did get referenced by some static variables, such as [cmd/annotate.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/annotate.go#L75), which will be impossible to decouple.

To make it easy, could we just replace the value `validResources` with a url, like <https://github.com/kubernetes/kubernetes.github.io/blob/master/docs/user-guide/kubectl-overview.md#resource-types>.

Reply:
We should not reference static docs, but add a command to list valid resources and short names for the current server instead (e.g. https://github.com/kubernetes/kubernetes/pull/42873) and reference that

Reply:
Yes.

But for [`cmd/annotate.go`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/annotate.go#L75), we could not make the request to fetch valid resources. Seems we can only modify the current long description.

Reply:
The long description should reference the command to run to obtain valid resources for the current live server

Reply:
Thanks @liggitt. Make sense. I will work this. 

But this PR is just for the documentation. We could update it this way, right?

Reply:
This documentation will *always* be wrong if we continue doing it this way.